packages/charts/src/components/legend/legend_icon.tsx

```diff
@@ -9,36 +9,43 @@
 import React from 'react';
 
 import { ShapeRendererFn } from '../../chart_types/xy_chart/renderer/shapes_paths';
-import { Color } from '../../utils/common';
-import { PointStyle } from '../../utils/themes/theme';
-import { Icon } from '../icons/icon';
+import { Color, isNil } from '../../utils/common';
+import { PointShape, PointStyle } from '../../utils/themes/theme';
 
 interface LegendIconProps {
   pointStyle?: PointStyle;
   color: Color;
 }
 
-const MARKER_SIZE = 16;
+const MARKER_SIZE = 8;
+
+// to limit size, set a min and max
+const getRatio = (radius: number | undefined, strokeWidth: number) => {
+  const adjustedRadius = isNil(radius) ? strokeWidth : ((MARKER_SIZE / 2) * strokeWidth) / (radius * 2);
+  if (1.5 > adjustedRadius) return 1.5;
+  return Math.min(adjustedRadius, (radius ?? strokeWidth) * 2);
+};
+
 /** @internal */
 export const LegendIcon = ({ pointStyle, color }: LegendIconProps) => {
-  if (!pointStyle || !pointStyle.shape) {
-    return <Icon type="dot" color={color} aria-label={`series color: ${color}`} />;
-  }
-  const { shape, fill, stroke, strokeWidth, opacity } = pointStyle;
-  const [shapeFn, rotation] = ShapeRendererFn[shape];
-  const adjustedSize = MARKER_SIZE - strokeWidth;
+  const { shape, fill, stroke, strokeWidth = 1, opacity, radius } = pointStyle ?? { radius: 4 };
+
+  const adjustedFill = shape ? fill ?? color : color;
+  const adjustedStrokeWidth = getRatio(radius, strokeWidth);
+  const [shapeFn, rotation] = ShapeRendererFn[shape ?? PointShape.Circle];
+  const adjustedSize = MARKER_SIZE - adjustedStrokeWidth;
   return (
-    <svg height={MARKER_SIZE} width={MARKER_SIZE} aria-label={`series color: ${color}`}>
+    <svg width={MARKER_SIZE * 2} height={MARKER_SIZE * 2} viewBox="0 0 16 16" aria-label={`series color: ${color}`}>
       <g
         transform={`
-            translate(${MARKER_SIZE / 2}, ${MARKER_SIZE / 2})
+            translate(${MARKER_SIZE}, ${MARKER_SIZE})
             rotate(${rotation})`}
       >
         <path
-          d={shapeFn(shape === 'triangle' || shape === 'plus' || shape === 'x' ? adjustedSize / 2 : adjustedSize / 3)}
+          d={pointStyle?.shape === PointShape.Diamond ? shapeFn(adjustedSize / 3) : shapeFn(adjustedSize / 2)}
           stroke={stroke ?? color}
```

Comment:
This should be assigned only when `shape` is defined

```suggestion
          stroke={shape ? stroke ?? color : undefined}
```

Reply:
80c4cea thanks!